apps/openmw/mwmechanics/recharge.cpp

```diff
@@ -0,0 +1,92 @@
+#include "recharge.hpp"
+
+#include <components/misc/rng.hpp>
+
+#include "../mwbase/world.hpp"
+#include "../mwbase/environment.hpp"
+#include "../mwbase/windowmanager.hpp"
+
+#include "../mwworld/containerstore.hpp"
+#include "../mwworld/class.hpp"
+#include "../mwworld/esmstore.hpp"
+
+#include "creaturestats.hpp"
+#include "actorutil.hpp"
+
+namespace MWMechanics
+{
+
+bool rechargeItem(const MWWorld::Ptr &item, const float maxCharge, const float duration)
+{
+    float charge = item.getCellRef().getEnchantmentCharge();
+    if (charge == -1 || charge == maxCharge)
+        return false;
+
+    static float fMagicItemRechargePerSecond = MWBase::Environment::get().getWorld()->getStore().get<ESM::GameSetting>().find(
+                "fMagicItemRechargePerSecond")->mValue.getFloat();
+
+    item.getCellRef().setEnchantmentCharge(std::min(charge + fMagicItemRechargePerSecond * duration, maxCharge));
+    return true;
+}
+
+bool rechargeItem(const MWWorld::Ptr &item, const MWWorld::Ptr &gem)
+{
+    if (!gem.getRefData().getCount())
+        return false;
+
+    MWWorld::Ptr player = MWMechanics::getPlayer();
+    MWMechanics::CreatureStats& stats = player.getClass().getCreatureStats(player);
+
+    float luckTerm = 0.1f * stats.getAttribute(ESM::Attribute::Luck).getModified();
+    if (luckTerm < 1 || luckTerm > 10)
+        luckTerm = 1;
```

Comment:
Since everything is being moved anyway, maybe copying vanilla oversights like this one should be reconsidered?